datasets/indosum/indosum.py

```diff
@@ -0,0 +1,163 @@
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        my_urls = _URL
+        data_dir = dl_manager.download_and_extract(my_urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepaths": [
+                        os.path.join(data_dir, "indosum", "train.01.jsonl"),
+                        os.path.join(data_dir, "indosum", "train.02.jsonl"),
+                        os.path.join(data_dir, "indosum", "train.03.jsonl"),
+                        os.path.join(data_dir, "indosum", "train.04.jsonl"),
+                        os.path.join(data_dir, "indosum", "train.05.jsonl"),
+                    ],
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepaths": [
+                        os.path.join(data_dir, "indosum", "test.01.jsonl"),
+                        os.path.join(data_dir, "indosum", "test.02.jsonl"),
+                        os.path.join(data_dir, "indosum", "test.03.jsonl"),
+                        os.path.join(data_dir, "indosum", "test.04.jsonl"),
+                        os.path.join(data_dir, "indosum", "test.05.jsonl"),
+                    ],
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepaths": [
+                        os.path.join(data_dir, "indosum", "dev.01.jsonl"),
+                        os.path.join(data_dir, "indosum", "dev.02.jsonl"),
+                        os.path.join(data_dir, "indosum", "dev.03.jsonl"),
+                        os.path.join(data_dir, "indosum", "dev.04.jsonl"),
+                        os.path.join(data_dir, "indosum", "dev.05.jsonl"),
+                    ],
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepaths, split):
+        """ Yields examples. """
+
+        logging.info("⏳ Generating examples from = %s", filepaths)
+        dataset = load_dataset("json", data_files=filepaths)
```

Comment:
Can you load the files using the `json` module here instead ?

Indeed the dataset scripts are not supposed to write files on disk and `load_dataset` does create cache files (arrow files).